scheduler/src/cook/mesos/mesos_mock.clj

```diff
@@ -361,22 +382,25 @@
 
    Parameters:
    `hosts` is a seq of maps that contain a hostname, agent id, resources and attributes
-   `speed-multiplier` is a number >1 which is how much to speed up 'time'
+   `offer-trigger-chan` chan that has an item put on it when offers should be sent
+   `scheduler` is an implementation of the mesomatic scheduler protocol
+   `state-atom` will reset atom after each cycle with updated state -- useful for debugging
    `task->runtime-ms` is a function that takes a task spec and returns the runtime in millis
    `task->complete-status` is a function that takes a task spec and returns a mesos complete status
    `scheduler` is an implementation of the mesomatic scheduler protocol
 
    Returns a mesos driver"
-  ([hosts speed-multiplier scheduler]
-   (mesos-mock hosts speed-multiplier default-task->runtime-ms scheduler))
-  ([hosts speed-multiplier task->runtime-ms scheduler]
-   (mesos-mock hosts speed-multiplier task->runtime-ms default-task->complete-status scheduler))
-  ([hosts speed-multiplier task->runtime-ms task->complete-status scheduler]
+  ([hosts offer-trigger-chan scheduler & 
+    {:keys [task->complete-status task->runtime-ms state-atom
+            complete-trigger-chan]
+     :or {task->complete-status default-task->complete-status
+          task->runtime-ms default-task->runtime-ms
+          state-atom (atom nil)
+          complete-trigger-chan (chime-ch (periodic/periodic-seq (t/now) (t/millis 20)))}}]
```

Comment:
Sort of, I was observing long delays for tasks to be completed because it would require many cycles for a bunch of tasks to complete around the same time to be completed. Also, moving to this model gives finer control over when completions happen (since we can trigger them).